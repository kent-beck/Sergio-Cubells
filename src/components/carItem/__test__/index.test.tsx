import { render, screen } from "@testing-library/react";
import CarItem from "..";
import { Car } from "../../../types";

describe("Renders a ", () => {
  const carItem: Car = {
    id: "0",
    title: "Coche",
    euroPrice: "29.870 €",
    image: "someImageUrl",
  };

  test("Car item", () => {
    const tree = render(<CarItem carItem={carItem} />).asFragment();
    expect(tree).toMatchSnapshot();
  });

  test("euro price text", () => {
    render(<CarItem carItem={carItem} />);

    const euroPriceTextElement = screen.getByText(/29.870 €/i);

    expect(euroPriceTextElement).toBeInTheDocument();
  });

  test("from text", () => {
    render(<CarItem carItem={carItem} />);

    const desdeTextElement = screen.getByText(/desde/i);

    expect(desdeTextElement).toBeInTheDocument();
  });

  test("title text", () => {
    render(<CarItem carItem={carItem} />);

    const titleTextElement = screen.getByText(/Coche/i);

    expect(titleTextElement).toBeInTheDocument();
  });

  test("alt image", () => {
    render(<CarItem carItem={carItem} />);

    const testImage = document.querySelector("img") as HTMLImageElement;

    expect(testImage.alt).toContain("Image for Coche");
  });

  test("src image", () => {
    render(<CarItem carItem={carItem} />);

    const testImage = document.querySelector("img") as HTMLImageElement;

    expect(testImage.src).toContain("http://localhost/someImageUrl");
  });
});
