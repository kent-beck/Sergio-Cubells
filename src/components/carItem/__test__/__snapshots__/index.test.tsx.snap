// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a  Car item 1`] = `
<DocumentFragment>
  <section
    class="carItemSection"
  >
    <div
      class="carItemContent"
    >
      <img
        alt="Image for Coche"
        class="carItemImage"
        src="someImageUrl"
      />
      <main
        class="carItemTextContainer"
      >
        <div
          class="carItemTitle"
        >
          Coche
        </div>
        <div
          class="carItemFrom"
        >
          desde
        </div>
        <div
          class="carItemPrice"
        >
          29.870 â‚¬
        </div>
      </main>
    </div>
  </section>
</DocumentFragment>
`;
